### ISSUE TITLE
Rustup nightly install failed for the 2019-05-11 update

### ISSUE THREAD

**Issue by maufirf** [rust-lang/rustup#1849]
**Problem**

I was trying to update my `rustup` to nigthly version with `rustup update nightly` to the latest, which happens to be `1.36.0-nightly (d595b1135 2019-05-10)`, but it said this instead:
```
error: failed to install component: 'rust-docs-x86_64-pc-windows-msvc', detected
 conflict: '"share/doc/rust/html"'
```

More in-depth console output:
```
info: syncing channel updates for 'nightly-x86_64-pc-windows-msvc'
info: latest update on 2019-05-11, rust version 1.36.0-nightly (d595b1135 2019-0
5-10)
info: downloading component 'rustc'
info: downloading component 'rust-std'
info: downloading component 'cargo'
info: downloading component 'rust-docs'
info: installing component 'rustc'
 60.5 MiB /  60.5 MiB (100 %)   7.8 MiB/s ETA:   0 s
info: installing component 'rust-std'
 55.2 MiB /  55.2 MiB (100 %)   9.2 MiB/s ETA:   0 s
info: installing component 'cargo'
  2.9 MiB /   2.9 MiB (100 %)   1.5 MiB/s ETA:   0 s
info: installing component 'rust-docs'
  2.1 MiB /  11.0 MiB ( 19 %)  16.0 KiB/s ETA: 9.513163248697916 min 30.78979492
  2.8 MiB /  11.0 MiB ( 26 %)  22.4 KiB/s ETA: 6.24749755859375 min 14.849853515
  2.8 MiB /  11.0 MiB ( 26 %)  19.2 KiB/s ETA: 7.281802707248264 min 16.90816243
  2.9 MiB /  11.0 MiB ( 26 %)  14.4 KiB/s ETA: 9.68129249855324 min 40.877549913
```
*Irrelevant part is cut as it repeats the same installing progress, you can see in comment history.*
```
  8.2 MiB /  11.0 MiB ( 75 %)   9.6 KiB/s ETA: 4.924716525607639 min 55.48299153
  8.3 MiB /  11.0 MiB ( 75 %)   8.0 KiB/s ETA: 5.809659830729166 min 48.57958984
  8.4 MiB /  11.0 MiB ( 76 %)  16.0 KiB/s ETA: 2.813163248697917 min 48.78979492
 11.0 MiB /  11.0 MiB (100 %) 260.8 KiB/s ETA:   0 s
info: rolling back changes
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\k_
oncg7yt_1u10dc_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\etc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\97
mi688spn10mdxc_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\etc/bash_completion.d'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\0u
owi_fmlyydq6sf_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\share/doc/cargo'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\fk
nq1e6suhpyhwzq_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\share/man'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\vi
fnxj91uuz9aa4z_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\share/man/man1'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\gz
iqtal1bvw0e990_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\share/zsh'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\q0
6wwmwfywgd180i_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\share/zsh/site-functions'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\zd
w0y3d47xyfx48r_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\ii
_0ndjz6ltychen_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\_1
eeszp07vykuqn__dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/lib'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\us
occ4ohyy34tllx_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\lib/rustlib/etc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\3z
z06qy6sngp_wx9_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\jw
004mq5wbtq52aa_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_6
4-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/codegen-backends'
error: failed to install component: 'rust-docs-x86_64-pc-windows-msvc', detected
 conflict: '"share/doc/rust/html"'
```

I am using Windows 8.0 and using CMD terminal. I tried it twice, in the VSCode terminal and directly in CMD (using administrator privilege for the latter), both yields the same result.

**Steps**

1. I set the directory of the terminal to my project folder, previously made using `cargo init`.
2. Simply typed `rustup update nightly`

**Possible Solution(s)**

*-*

**Notes**

Output of `rustup --version`: `rustup 1.17.0 (069c88ed6 2019-03-05)`
Output of `rustup show`:
```
Default host: x86_64-pc-windows-msvc

installed toolchains
--------------------

stable-x86_64-pc-windows-msvc
nightly-x86_64-pc-windows-msvc

active toolchain
----------------

stable-x86_64-pc-windows-msvc (default)
rustc 1.33.0 (2aa4c46cf 2019-02-28)
```
I am using Windows 8.0.
This problem is encountered when following the instructions to set up the IntermezzOS in [its book](https://intermezzos.github.io/book/second-edition/hello-world/setup.html#stable-vs-nightly-rust)
Command Prompt is run both with and without administrator privilege.

**Comment by kinnison** [comment]
Hi,

Nightly can sometimes have problems like this.  Could you confirm if this remains the case in a few days?

If it does, then we might have something to investigate.

Thanks,

Daniel.

**Comment by maufirf** [comment]
Hi

So i have tried it again today, same problem still presist, here's the console output:
```
info: syncing channel updates for 'nightly-x86_64-pc-windows-msvc'
info: latest update on 2019-05-15, rust version 1.36.0-nightly (372be4f36 2019-05-14)
info: downloading component 'rustc'
 60.6 MiB /  60.6 MiB (100 %)   1.1 MiB/s ETA:   0 s                n 1.8309497833251953 s
info: downloading component 'rust-std'
 55.3 MiB /  55.3 MiB (100 %)   1.2 MiB/s ETA:   0 s
info: downloading component 'cargo'
  2.9 MiB /   2.9 MiB (100 %) 721.7 KiB/s ETA:   0 s
info: downloading component 'rust-docs'
 11.0 MiB /  11.0 MiB (100 %) 883.0 KiB/s ETA:   0 s
info: installing component 'rustc'
 60.6 MiB /  60.6 MiB (100 %)   2.4 MiB/s ETA:   0 s                n 53.821121717282864 s
info: installing component 'rust-std'
 55.3 MiB /  55.3 MiB (100 %)   7.6 MiB/s ETA:   0 s
info: installing component 'cargo'
  2.9 MiB /   2.9 MiB (100 %)   1.5 MiB/s ETA:   0 s
info: installing component 'rust-docs'
 11.0 MiB /  11.0 MiB (100 %) 240.0 KiB/s ETA:   0 s                n 50.54638671875 s
info: rolling back changes
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\d3u_nbuskqgbfxfh_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/zsh'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\nitsuf2puwxx1rdh_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/zsh/site-functions'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\5ds4m_rr3yc5xyur_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/man'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\rot4574qpm6g5bqs_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/man/man1'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\s08vpr9ca5uv8089_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/doc/cargo'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\yuon_z996rcxks___dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\etc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\y2zlizxlju8fh84c_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\etc/bash_completion.d'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\6zj9fieb1lu0f8ze_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\sfkfb_ougoh0gcb2_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\s89yu3rlw20n3iz3_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/lib'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\hai5lwntkare5a6w_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\78uxkxsuefv48hjb_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/etc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\2t1y_0hu4ubpsdbe_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/codegen-backends'
error: failed to install component: 'rust-docs-x86_64-pc-windows-msvc', detected conflict: '"share/doc/rust/html"'
```

**Comment by kinnison** [comment]
Okay, thank you @parampaa2 for trying again.

So now we know there's something we need to diagnose, let's step another step along the path.

Based on the log you're showing, I think you're running a slightly old copy of rustup, so let's eliminate that as a problem by first running `rustup self update` before trying `rustup update` once more.

Also, can you confirm how much free space you have on the C: partition, and whether you are running any anti-virus beyond Windows Defender?

**Comment by maufirf** [comment]
Alright, so i have recently updated the rustup using `rustup self update`, and it was a success with this output:
```
info: checking for self-updates
info: downloading self-update
info: rustup updated successfully to 1.18.2
```

However, it still does not relieve the problem, using `rustup update nightly` yields the same result:
```
info: syncing channel updates for 'nightly-x86_64-pc-windows-msvc'
381.3 KiB / 381.3 KiB (100 %)  44.6 KiB/s ETA:   0 s
info: latest update on 2019-05-16, rust version 1.36.0-nightly (7158ed9cb 2019-05-15)
info: downloading component 'rustc'
 60.6 MiB /  60.6 MiB (100 %) 337.6 KiB/s ETA:   0 s
info: downloading component 'rust-std'
 55.3 MiB /  55.3 MiB (100 %) 347.1 KiB/s ETA:   0 s
info: downloading component 'cargo'
  2.9 MiB /   2.9 MiB (100 %) 411.7 KiB/s ETA:   0 s
info: downloading component 'rust-docs'
 11.0 MiB /  11.0 MiB (100 %) 669.4 KiB/s ETA:   0 s
info: installing component 'rustc'
 60.6 MiB /  60.6 MiB (100 %)   1.8 MiB/s ETA:   0 s
info: installing component 'rust-std'
 55.3 MiB /  55.3 MiB (100 %)   4.5 MiB/s ETA:   0 s
info: installing component 'cargo'
  2.9 MiB /   2.9 MiB (100 %)   1.5 MiB/s ETA:   0 s
info: installing component 'rust-docs'
 11.0 MiB /  11.0 MiB (100 %) 254.4 KiB/s ETA:   0 s
info: rolling back changes
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\0lrta43etg8q_uvc_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/zsh'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\7a0u8cad801j4f6b_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/zsh/site-functions'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\60ug_pulb8orob_5_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\etc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\99lny0j1fqwve0sl_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\etc/bash_completion.d'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\iwarmgwoys44ede__dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/doc/cargo'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\4rn6a1yphvu3utkp_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/man'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\t0t7yv20omyfj6vu_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/man/man1'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\mlb_o0mtq_27p7mp_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\7d0n7j_v69dknj4k_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\cjm91p2z3py4ewin_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/lib'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\gpobkv2kkypodmve_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/codegen-backends'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\b5_hqzgla2akwah5_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\3qia6ol22m5jzew6_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/etc'
error: failed to install component: 'rust-docs-x86_64-pc-windows-msvc', detected conflict: '"share/doc/rust/html"'
```

I recently backed up my personal data from my C: partition, which is 41 GB right now. Previously, on the first encounter of the problem, i have 33 GB.

Also as an additional information, 2 (out of 36) of my friends claimed they successfully executing `rustup update nightly`, I will ask their computer details. And also, i once tried this on one of the computers in my faculty computer lab (Using Win10, First rust installation is about one or two month ago, and has VS 2017 with C++ package installed) and the `rustup update nightly` went succeeded.

An addition to my current Win8.0 PC, i used VS 2015. I am running AVG antivirus, disabling it yields the same result:
```
info: syncing channel updates for 'nightly-x86_64-pc-windows-msvc'
381.3 KiB / 381.3 KiB (100 %)  41.2 KiB/s ETA:   0 s
info: latest update on 2019-05-16, rust version 1.36.0-nightly (7158ed9cb 2019-05-15)
info: downloading component 'rustc'
info: downloading component 'rust-std'
info: downloading component 'cargo'
info: downloading component 'rust-docs'
info: installing component 'rustc'
 60.6 MiB /  60.6 MiB (100 %)   3.1 MiB/s ETA:   0 s
info: installing component 'rust-std'
 55.3 MiB /  55.3 MiB (100 %)   1.9 MiB/s ETA:   0 s
info: installing component 'cargo'
info: installing component 'rust-docs'
 11.0 MiB /  11.0 MiB (100 %) 420.8 KiB/s ETA:   0 s
info: rolling back changes
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\8q7yfn29ohs0gstb_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/man'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\sgbeo2yfot3hzz0p_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/man/man1'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\zzfgil6joj5kg5ub_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\3iic0q19v3cps6hz_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/doc/cargo'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\c0lyrb4yzvf3sgwf_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\etc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\bjeiq65lhzvb73wm_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\etc/bash_completion.d'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\cb85tf1xe6icittf_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/zsh'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\x1zl466u2y1faxv9_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\share/zsh/site-functions'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\lrkwf0btnpat58pu_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\x9tzffk_se6hjj8d_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/lib'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\j5x_fgvqh2ba3e83_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/codegen-backends'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\il8dy6vkd900k0vo_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/x86_64-pc-windows-msvc/bin'
error: could not rename component file from 'C:\Users\Aufi Rayesa\.rustup\tmp\f_i62v905oiyo_pk_dir\bk' to 'C:\Users\Aufi Rayesa\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/etc'
error: failed to install component: 'rust-docs-x86_64-pc-windows-msvc', detected conflict: '"share/doc/rust/html"'
```

Thanks for your assistance, i hope these helped.

**Comment by kinnison** [comment]
Hi,

Thanks for that, it has eliminated some paths of investigation, for sure.

So what we're left with is a question of whether it could be a Windows-specific issue, perhaps related to path lengths, or a rustup issue with the space in your username, or something else.

@rbtcollins Do you happen to have any ideas on this issue?  Would it help if we had a zip of this user's `.rustup` to look at?

**Comment by rbtcollins** [comment]
The errors being logged are all during the rollback - the bk directory is for rollbacks. We're not seeing the actual error other than 'detected conflict'.

Thats a ComponentConflict, which only gets emitted from dist::component::transaction::ChangedItem.dest_abs_path.

add file, copy file, copy dir, move file, move dir call that.

dest_abs_path stats the destination and if it exists, triggers the error.
```
        if utils::path_exists(&abs_path) {
            Err(ErrorKind::ComponentConflict {
                name: component.to_owned(),
                path: relpath.clone(),
            }
            .into())
```

I believe the logic is this:
1) move the existing files to backup
2) unpack and move into place
3) delete backups

So this is failing at 2 because something is causing "share/doc/rust/html" to still exist in the target directory.

I can think of a few possilbilities; 
1) perhaps rust-docs was removed from the metadata (so its not being backed up) but it still on disk (so we get a conflict). Or some variation.
2) perhaps the file system is returning from the move call in 1 but stil reporting the files presence in the stat during 2. possibilities including network file system caching layers, or the weird thing I've read about but not dug into where delete() can return but the file still exists for a bit - perhaps that happens on move too, but we just haven't actually observed it previously.

I'd start by looking at the rustup metadata to see if it thinks rustc-docs is installed in that toolchain presently.

**Comment by rbtcollins** [comment]
I should mention, I haven't looked closely at the local data store details,
so modify my suggested investigation path based on your better knowledge.
Tldr though is that we should make sure rust knows there is content at the
conflicting path before the update begins, and that it is moving it out of
the way, before suspecting platform issues.

Perhaps running with -v will help.

**Comment by kinnison** [comment]
My working hypothesis here, since I've now seen it elsewhere, is that at least one of the manifest files in `lib/rustlib/` under the toolchain has been lost.

**Comment by kinnison** [comment]
Hi @parampaa2 

Could you tell me if you're still having this problem, or if you resolved it?  If you're still having it, could you tell me if any of the files called `manifest-...` in the toolchain directory under `lib/rustlib` exist but are empty (zero size) ?

If you run `rustup show` it'll tell you where Rustup's home is.  Look in the toolchains directory in there.

**Comment by rbtcollins** [comment]
This is another instance of missing/corrupt manifest data. Without knowledge of the installed file, rustup cannot remove it prior to installing the updated version, and then hits a conflict due to the file already existing, and you get the error.

We've got a bunch of these bugs scattered all over the place, I'm closing them all to bring order. See #2417.
